dev/com.ibm.ws.logging/src/com/ibm/ws/logging/data/LogTraceData.java

```diff
@@ -357,7 +362,20 @@ public long getDatetime() {
     }
 
     public String getMessageId() {
-        return getStringValue(1);
+        if (getStringValue(1) == null || getStringValue(1).isEmpty()) {
+            String message = getMessage();
+            String messageId = "";
+            if (message != null) {
+                messageId = parseMessageId(message);
+                setPair(1, messageId);
+                return messageId;
+            } else {
+                setPair(1, "");
+                return messageId;
+            }
+        } else {
+            return getStringValue(1);
+        }
```

Comment:
for better performance, should call getStringValue(1) once and keep it in a variable, then reuse it throughout the rest of this method.